### PR TITLE
Split destroy to plan and apply

### PR DESCRIPTION
**Pull request by nsarupr**
# Description

# Safety checklist
* [x] This change is backwards compatible and safe to apply by existing users
* [x] This change will NOT lead to data loss
* [x] This change will NOT lead to downtime who already has an env/service setup

## How has this change been tested, beside unit tests?
YOUR_ANSWER

### CODE REVIEW

File: opta/commands/destroy.py [hunk 1]
```diff
@@ -105,8 +117,36 @@ def destroy(
     layer.modules = [x for x in layer.modules if x.name in modules]
     gen_all(layer)
     Terraform.init(False, "-reconfigure", layer=layer)
-    logger.info(f"Destroying {layer.name}")
-    Terraform.destroy_all(layer, *tf_flags)
```
Comment (juandiegopalomino) [inline opta/commands/destroy.py:109]: should we get rid of destroy_all?

Comment (nsarupr) [reply opta/commands/destroy.py:109]: I thought it better to have the `plan` and `apply` in `destroy.py` just the way it was in `apply.py`, and replicated the structure but with `-destroy` flag.
Also the state deletion which is happening inside `destroy_all` has been made as a separate method.
Is there anything that I might have missed?

Comment (juandiegopalomino) [reply opta/commands/destroy.py:109]: What I meant is that I don't think that the destroy_all function is used for anything else, so we should delete it

Comment (nsarupr) [reply opta/commands/destroy.py:109]: My bad. I understood different, but yeah, we should get rid of the `destroy_all` method.

**Review by juandiegopalomino — APPROVED** [review]

**Pull request merged**